openlibrary/catalog/add_book/tests/test_add_book.py

```diff
@@ -1179,3 +1184,52 @@ def test_add_identifiers_to_edition(mock_site) -> None:
     e = mock_site.get(reply['edition']['key'])
     assert e.works[0]['key'] == '/works/OL19W'
     assert e.identifiers._data == {'goodreads': ['1234'], 'librarything': ['5678']}
+
+
+@pytest.mark.parametrize(
+    'name,rec,error,expected',
+    [
+        (
+            "Books that are too old can't be imported",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '1499'},
+            PublicationYearTooOld,
+            None,
+        ),
+        (
+            "But 1500 CE+ can be imported",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '1500'},
+            None,
+            None,
+        ),
+        (
+            "But trying to import a book from a future year raises an error",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '3000'},
+            PublishedInFutureYear,
+            None,
+        ),
+        (
+            "Independently published books can't be imported",
+            {
+                'title': 'a book',
+                'source_records': ['ia:ocaid'],
+                'publishers': ['Independently Published'],
+            },
+            IndependentlyPublished,
+            None,
+        ),
+        (
+            "Can't import sources that require an ISBN",
+            {'title': 'a book', 'source_records': ['amazon:amazon_id'], 'isbn_10': []},
+            SourceNeedsISBN,
+            None,
+        ),
+    ],
+)
+def test_validate_record(name, rec, error, expected) -> None:
+    _ = name  # Name is just used to make the tests easier to understand.
+
+    if error:
+        with pytest.raises(error):
+            validate_record(rec)
+    else:
+        assert validate_record(rec) is expected  # type: ignore [func-returns-value]
```

Comment:
`name` could be used here if the assertation isn't met in the assert failed message.

Reply:
@hornc could you elaborate? I'm not even sure that passing `name` is useful here.

Reply:
@hornc, I updated the PR to reflect this suggestion. E.g.:
```
>       assert sorted(get_missing_fields(rec=rec)) == sorted(expected), f"Assertion failed for test: {name}"                                                                                                                                                                                
E       AssertionError: Assertion failed for test: Catches multiple missing required fields                                                                                                                                                                                                 
E       assert ['source_records', 'title'] == ['flop', 'source_records']                                                                                                                                                                                                                    
E         At index 0 diff: 'source_records' != 'flop'                                                                                                                                                                                                                                       
E         Use -v to get more diff                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                            
openlibrary/tests/catalog/test_utils.py:338: AssertionError
```